src/approaches/nsrt_rl_approach.py

```diff
@@ -49,33 +65,215 @@ def get_interaction_requests(self) -> List[InteractionRequest]:
             requests.append(request)
         return requests
 
+    @staticmethod
+    def _get_unchanging_features(
+            changing_var_to_feat: Dict[Variable, List[int]],
+            var_to_obj: VarToObjSub) -> Dict[Variable, List[int]]:
+        var_to_unchanging_feat_ind = {}
+        for var, changing_indices in changing_var_to_feat.items():
+            dim = var_to_obj[var].type.dim
+            unchanging_indices = [
+                i for i in range(dim) if i not in changing_indices
+            ]
+            var_to_unchanging_feat_ind[var] = unchanging_indices
+        return var_to_unchanging_feat_ind
+
     def learn_from_interaction_results(
             self, results: Sequence[InteractionResult]) -> None:
         self._online_learning_cycle += 1
         # We get one result per training task.
         for i, result in enumerate(results):
-            states = result.states
-            actions = result.actions
-            traj = LowLevelTrajectory(states,
-                                      actions,
+            traj = LowLevelTrajectory(result.states,
+                                      result.actions,
                                       _is_demo=False,
                                       _train_task_idx=i)
-            self._online_trajectories.append(traj)
+            self._train_task_to_online_traj[i] = traj
+
+        # This maps each unique _Option to its experience data from the previous
+        # online learning cycle. The experience data is a List of (states,
+        # actions, rewards, relative_parameters) tuples. Each tuple contains
+        # information from a single instance the _Option was run. Each element
+        # of the tuple is a list: 'states' is the list of states visited,
+        # 'actions' is the list of actions taken, 'rewards' is the list of the
+        # rewards acculumated, and 'relative_params' is a list of (relative)
+        # parameter that was passed into the _LearnedNeuralParameterizedOption's
+        # regressor. The relative parameter is necessary to store because during
+        # learning, the RLOptionLearner will need to reconstruct the input to
+        # the regressor.
+        option_to_data: Dict[str,
+                             List] = {}  # option_name -> online experience
+
+        # For each training task, compute the experience data for each _Option
+        # we see used in that training task.
+        for i in range(len(self._train_tasks)):
+            plan = self._train_task_to_option_plan[i]
+            traj = self._train_task_to_online_traj[i]
+
+            curr_option_idx = 0
+            curr_option = plan[curr_option_idx]
+            parent = cast(_LearnedNeuralParameterizedOption,
+                          curr_option.parent)
+            var_to_obj = dict(
+                zip(parent._operator.parameters, curr_option.objects))  # pylint: disable=protected-access
+            var_to_unchanging_feat_ind = \
```

Comment:
I'm hoping we can get rid of this line based on my other comments, but just FYI, in this situation, you can do `self._get_unchanging_features` instead of `NSRTReinforcementLearningApproach._get_unchanging_features`